meta-refkit/classes/image-installer.bbclass

```diff
@@ -0,0 +1,370 @@
+
+                    get_option () (
+                        number=$1; shift
+                        i=0
+                        for option in "$@"; do
+                            if [ $i -eq "$number" ]; then
+                               echo "$option"
+                               return 0
+                            fi
+                            i=$( expr $i + 1 )
+                        done
+                        error "$answer is not a valid shortcut number, try again."
+                        return 1
+                    )
+
+                    read answer
+                    if [ ! "$answer" ]; then
+                        get_option 0 "$@" && return
+                    elif echo "$answer" | grep -q -e '^[0-9]*$'; then
+                        # Might be an invalid number, so do not return unconditionally!
+                        get_option "$answer" "$@" && return
+                    else
+                        echo "$answer"
+                        return
+                    fi
+                fi
+            done
+        fi
+    )
+
+    confirm_install () (
+        if ! istrue FORCE; then
+            while true; do
+                printf "%s" 'Proceed? Type "yes" to confirm and "no" to abort: '
+                read answer
+                case $answer in
+                    yes) break;;
+                    no) info "Aborting as requested."; return 1;;
+                esac
+            done
+        fi
+    )
+}
+
+INSTALLER_PICK_INPUT[shellcheck] = "sh"
+INSTALLER_PICK_INPUT () {
+    # We know what's going to be available, therefore we don't need code which looks
+    # for images.
+    AVAILABLE_INPUT="${@ ' '.join(['%s-${MACHINE}.%s' % (image, suffix) for image, suffix in [x.split(':') for x in d.getVar('INSTALLER_SOURCE_IMAGES').split()]])}"
+    CHOSEN_INPUT="${CHOSEN_INPUT:-}"
+
+    pick_input () {
+        # shellcheck disable=SC2086
+        CHOSEN_INPUT=$(pick_option "ambiguous input images" "Pick an image file" "$CHOSEN_INPUT" $AVAILABLE_INPUT) || exit 1
+    }
+}
+
+
+INSTALLER_FIND_OUTPUT[shellcheck] = "sh"
+INSTALLER_FIND_OUTPUT () {
+    AVAILABLE_OUTPUT=""
+
+    _find_output () (
+        # Find block devices and filter out our boot device and read-only devices.
+        local root_device=$(findmnt / --output SOURCE --noheadings)
+        # Might have been a symlink.
+        root_device=$(readlink -f "$root_device")
+        info "Boot device is $root_device."
+        local exclude=
+        if [ "$root_device" ]; then
+            # lsblk knows about the device topology and orders accordingly
+            # when the NAME column is enabled. Here we are relying on the
+            # fact that dependent items appear below the disk they are
+            # rooted in.
+            local disk=
+            # shellcheck disable=SC2034
+            exclude=$( lsblk --output NAME,KNAME,TYPE --ascii --noheadings | while read name kname type; do
```

Comment:
That's for the target, covered by:

    INSTALLER_RDEPENDS ??= " \
        util-linux \
    "